torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -127,13 +127,17 @@ TensorPipeAgent::TensorPipeAgent(
 }
 
 TensorPipeAgent::~TensorPipeAgent() {
+  VLOG(1) << "RPC agent for " << workerInfo_.name_ << " is being destroyed";
```

Comment:
I added it because at some point I was investigating shutdown-related issues (see also #39663). As the current RPC agent is stored in shared_ptrs the moment it's destructed isn't necessarily easy to predict, as any user holding on their shared_ptr can delay it. I think it may make sense to have this information when debugging that type of issues.